beacon-chain/rpc/beacon_chain_server.go

```diff
@@ -206,12 +208,47 @@ func (bs *BeaconChainServer) ListValidatorAssignments(
 	return nil, status.Error(codes.Unimplemented, "not implemented")
 }
 
-// GetValidatorParticipation retrieves the validator participation information for a given epoch.
+// GetValidatorParticipation retrieves the validator participation information for a given epoch,
+// it returns the information about validator's participation rate
 //
-// This method returns information about the global participation of
-// validator attestations.
+// TODO(#3064): Implement validator participation for a specific epoch. Current implementation returns latest,
+// this is blocked by DB refactor.
 func (bs *BeaconChainServer) GetValidatorParticipation(
 	ctx context.Context, req *ethpb.GetValidatorParticipationRequest,
 ) (*ethpb.ValidatorParticipation, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+
+	s, err := bs.beaconDB.HeadState(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not retrieve current state: %v", err)
+	}
+	finalizedBlk, err := bs.beaconDB.FinalizedBlock()
```

Comment:
cant you get the finalized epoch from the state ? why retrieve the finalized block too

Reply:
Yea, this will get changed once add filter for state retrieval in db. I think getting the finalized epoch from state is ok for now

Reply:
done!